sale_coupon_advanced/models/sale_order.py

```diff
@@ -6,15 +6,30 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _get_next_order_line_sequence(self):
+        self.ensure_one()
+        seq = 1
+        # Expecting non empty list.
+        sequences = self.mapped("order_line.sequence")
+        try:
+            return max(sequences) + 1
+        # Will get ValueError, if sequences is empty list.
+        except ValueError:
+            return seq
+
     def add_reward_line_values(self, program):
         """Add the rewarded product if a reward line has been found for this
         product
         """
+        # NOTE. This method dosctring is misleading, it says, it adds
```

Comment:
you can update docstring with the real work method doing

Reply:
I thought first to highlight it for review. But you are right, I guess I can just update myself.